test/unit/directory_loader_test.lua

```diff
@@ -16,9 +17,15 @@ g.test_missing_folder = function()
     t.assert_error_msg_contains('is not valid', loader.list, loader)
 end
 
-g.test_empty_folder = function()
-    local list = loader.new('test/unit/migrations/empty'):list()
-    t.assert_equals(list, { })
+g.test_dynamic_folder = function()
+    fio.rmtree('test/unit/migrations/empty')
+    fio.mkdir('test/unit/migrations/empty')
```

Comment:
Maybe `fio.tempdir()` is better instead of calling `rm -rf` in a git tree?

Reply:
Directory loader accepts (intentionally) paths relative to searchroot, so that migrations would always be stored alongside application code